modules/k8s/deployment.go

```diff
@@ -98,13 +98,16 @@ func WaitUntilDeploymentAvailableE(
 
 // IsDeploymentAvailable returns true if all pods within the deployment are ready and started
 func IsDeploymentAvailable(deploy *appsv1.Deployment) bool {
-	for _, dc := range deploy.Status.Conditions {
-		if dc.Type == appsv1.DeploymentProgressing &&
-			dc.Status == v1.ConditionTrue &&
-			dc.Reason == "NewReplicaSetAvailable" {
-			return true
+	dc := getDeploymentCondition(deploy, appsv1.DeploymentProgressing)
+	return dc != nil && dc.Status == v1.ConditionTrue && dc.Reason == "NewReplicaSetAvailable"
```

Comment:
I was wondering if it's possible to implement a test case that would generate such a situation.

Reply:
Which situation are you referring to? The panic?
Won't be possible to reproduce with a real K8s apiserver. However, I just updated the commit to add unit tests.